### PR TITLE
Nickname database

### PR DESCRIPTION
**Pull request by veken1199**

### CODE REVIEW

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (g-harel) [inline src/org/thoughtcrime/securesms/NicknameHandler.java]: Does this first sentence really say anything that cannot be seen instantly?

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (g-harel) [inline src/org/thoughtcrime/securesms/NicknameHandler.java]: Should this be

```diff
+ return this.recipientDatabase.setNickname(recipient, nickname);
- return recipientDatabase.setNickname(recipient, nickname);
```

?

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (g-harel) [inline src/org/thoughtcrime/securesms/NicknameHandler.java]: This means we cannot merge yet?

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (g-harel) [inline src/org/thoughtcrime/securesms/NicknameHandler.java:8]: extra line

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (g-harel) [inline src/org/thoughtcrime/securesms/NicknameHandler.java]: this. ?

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (g-harel) [inline src/org/thoughtcrime/securesms/NicknameHandler.java]: Same comment about the first sentence.

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (g-harel) [inline src/org/thoughtcrime/securesms/NicknameHandler.java]: this. ?

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (g-harel) [inline src/org/thoughtcrime/securesms/NicknameHandler.java]: I don't understand the point of this whole class. It seems to be some kind of database helper, but right now it does almost nothing.

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (g-harel) [inline src/org/thoughtcrime/securesms/NicknameHandler.java]: If the other methods call this, does it need to be public? Would make using the class a bit easier.

File: src/org/thoughtcrime/securesms/database/DatabaseFactory.java
Comment (g-harel) [inline src/org/thoughtcrime/securesms/database/DatabaseFactory.java]: nickname is one word

File: src/org/thoughtcrime/securesms/database/DatabaseFactory.java
Comment (g-harel) [inline src/org/thoughtcrime/securesms/database/DatabaseFactory.java]: Is this the `"null"` string or `NULL` ?

File: src/org/thoughtcrime/securesms/database/RecipientDatabase.java
Comment (g-harel) [inline src/org/thoughtcrime/securesms/database/RecipientDatabase.java]: nickname

File: src/org/thoughtcrime/securesms/database/RecipientDatabase.java
Comment (g-harel) [inline src/org/thoughtcrime/securesms/database/RecipientDatabase.java]: .

File: src/org/thoughtcrime/securesms/database/RecipientDatabase.java
Comment (g-harel) [inline src/org/thoughtcrime/securesms/database/RecipientDatabase.java]: Is there a line limit in the coding style guide?

File: src/org/thoughtcrime/securesms/database/RecipientDatabase.java
Comment (g-harel) [inline src/org/thoughtcrime/securesms/database/RecipientDatabase.java]: lol `true ? true : false;`

File: src/org/thoughtcrime/securesms/database/RecipientDatabase.java
Comment (g-harel) [inline src/org/thoughtcrime/securesms/database/RecipientDatabase.java]: I feel like this method call can be formatted a bit nicer.

File: src/org/thoughtcrime/securesms/database/RecipientDatabase.java
Comment (g-harel) [inline src/org/thoughtcrime/securesms/database/RecipientDatabase.java]: nickname

File: src/org/thoughtcrime/securesms/database/RecipientDatabase.java
Comment (g-harel) [inline src/org/thoughtcrime/securesms/database/RecipientDatabase.java]: .

File: test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java
Comment (g-harel) [inline test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java]: .

File: test/unitTest/java/org/thoughtcrime/securesms/database/NicknameTests.java
Comment (g-harel) [inline test/unitTest/java/org/thoughtcrime/securesms/database/NicknameTests.java]: .

File: test/unitTest/java/org/thoughtcrime/securesms/database/NicknameTests.java
Comment (g-harel) [inline test/unitTest/java/org/thoughtcrime/securesms/database/NicknameTests.java]: .

File: test/unitTest/java/org/thoughtcrime/securesms/database/NicknameTests.java
Comment (g-harel) [inline test/unitTest/java/org/thoughtcrime/securesms/database/NicknameTests.java]: O_O

File: test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java
Comment (g-harel) [inline test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java]: should do this before merge

File: test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java
Comment (g-harel) [inline test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java]: this one is not used anywhere.

File: test/unitTest/java/org/thoughtcrime/securesms/nickname/NicknameHandlerTests.java
Comment (g-harel) [inline test/unitTest/java/org/thoughtcrime/securesms/nickname/NicknameHandlerTests.java]: .

File: test/unitTest/java/org/thoughtcrime/securesms/nickname/NicknameHandlerTests.java
Comment (g-harel) [inline test/unitTest/java/org/thoughtcrime/securesms/nickname/NicknameHandlerTests.java]: ...............................................................................................................................................................

**Review by g-harel — DISMISSED** [review]

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (veken1199) [reply src/org/thoughtcrime/securesms/NicknameHandler.java]: Yes, it tells you that This method allows the developer to set the underlining database, it is important to note this point as long as we are using javadoc style

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (veken1199) [reply src/org/thoughtcrime/securesms/NicknameHandler.java]: Yes, your approach is better in terms of readability

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (veken1199) [reply src/org/thoughtcrime/securesms/NicknameHandler.java]: We can merge, actuall @Dan-Yoo  needs to continue working from this code from his branch

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (veken1199) [reply src/org/thoughtcrime/securesms/NicknameHandler.java:8]: noted

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (veken1199) [reply src/org/thoughtcrime/securesms/NicknameHandler.java]: sure

File: test/unitTest/java/org/thoughtcrime/securesms/database/NicknameTests.java
Comment (veken1199) [reply test/unitTest/java/org/thoughtcrime/securesms/database/NicknameTests.java]: magicccccc

File: test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java
Comment (veken1199) [reply test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java]: no, we need to merge, so that @Dan-Yoo starts working on his "remove" feature and write his own tests

File: test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java
Comment (veken1199) [reply test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java]: what do you mean?
they are, maybe not recipient2

File: test/unitTest/java/org/thoughtcrime/securesms/nickname/NicknameHandlerTests.java
Comment (veken1199) [reply test/unitTest/java/org/thoughtcrime/securesms/nickname/NicknameHandlerTests.java]: ok

File: test/unitTest/java/org/thoughtcrime/securesms/nickname/NicknameHandlerTests.java
Comment (veken1199) [reply test/unitTest/java/org/thoughtcrime/securesms/nickname/NicknameHandlerTests.java]: ????

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (g-harel) [reply src/org/thoughtcrime/securesms/NicknameHandler.java]: It is literally called `SET database`

File: test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java
Comment (g-harel) [reply test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java]: So remove it from here ?

File: test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java
Comment (g-harel) [reply test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java]: `recipient2` is not used anywhere

File: test/unitTest/java/org/thoughtcrime/securesms/nickname/NicknameHandlerTests.java
Comment (g-harel) [reply test/unitTest/java/org/thoughtcrime/securesms/nickname/NicknameHandlerTests.java]: extra lines again

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (g-harel) [reply src/org/thoughtcrime/securesms/NicknameHandler.java]: So remove this from your branch then ?

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (veken1199) [reply src/org/thoughtcrime/securesms/NicknameHandler.java]: this method allows us to write tests and check if nickname feature works, if we dont do this, fragment will right away call databases and we can not test fragments

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (veken1199) [reply src/org/thoughtcrime/securesms/NicknameHandler.java]: yes but the rest of the comment is important in my opinion,

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (veken1199) [reply src/org/thoughtcrime/securesms/NicknameHandler.java]: again, since it is javadoc, I was thinking to write something that is very clear and explains the reason for this methods existence

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (veken1199) [reply src/org/thoughtcrime/securesms/NicknameHandler.java]: I dont understand

File: src/org/thoughtcrime/securesms/database/DatabaseFactory.java
Comment (veken1199) [reply src/org/thoughtcrime/securesms/database/DatabaseFactory.java]: sure

File: src/org/thoughtcrime/securesms/database/DatabaseFactory.java
Comment (veken1199) [reply src/org/thoughtcrime/securesms/database/DatabaseFactory.java]: I had a hard time setting db attr to NULL using update statement since they always set it to "NULL" so i decided to make the default "null"  since java knows null but not NULL for some reason,

File: src/org/thoughtcrime/securesms/database/RecipientDatabase.java
Comment (veken1199) [reply src/org/thoughtcrime/securesms/database/RecipientDatabase.java]: sure

Comment (veken1199) [reply src/org/thoughtcrime/securesms/database/RecipientDatabase.java]: but the db attribute is set to nick_name, is that ok ?

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (veken1199) [reply src/org/thoughtcrime/securesms/NicknameHandler.java]: I think it is nice to keep it to guide him and he ll know what is missing and from where he wants to start. But maybe we can rename this branch to "nickname_backend" and will not be merged until everything is finished.. @Dan-Yoo what do you think ?

File: src/org/thoughtcrime/securesms/database/RecipientDatabase.java
Comment (veken1199) [reply src/org/thoughtcrime/securesms/database/RecipientDatabase.java]: sure

File: src/org/thoughtcrime/securesms/database/RecipientDatabase.java
Comment (veken1199) [reply src/org/thoughtcrime/securesms/database/RecipientDatabase.java]: no

File: src/org/thoughtcrime/securesms/database/RecipientDatabase.java
Comment (veken1199) [reply src/org/thoughtcrime/securesms/database/RecipientDatabase.java]: where do you read the first true?

File: src/org/thoughtcrime/securesms/database/RecipientDatabase.java
Comment (veken1199) [reply src/org/thoughtcrime/securesms/database/RecipientDatabase.java]: can you propose something ??? this method will also be used by removeNickname, and the logic that it implements is that if the new nickname is the same as the old nickname then return false. otherwise return true

File: src/org/thoughtcrime/securesms/database/RecipientDatabase.java
Comment (veken1199) [reply src/org/thoughtcrime/securesms/database/RecipientDatabase.java]: sure

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (veken1199) [reply src/org/thoughtcrime/securesms/NicknameHandler.java]: yes but this method only used in testing

Comment (g-harel) [reply src/org/thoughtcrime/securesms/NicknameHandler.java]: I agree the second part is relevant, that's why I said first sentence in my original comment.

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (g-harel) [reply src/org/thoughtcrime/securesms/NicknameHandler.java:8]: It's still there

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (g-harel) [reply src/org/thoughtcrime/securesms/NicknameHandler.java]: it is clear without the need of the first sentence

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (g-harel) [reply src/org/thoughtcrime/securesms/NicknameHandler.java]: But couldn't the "setDatabase" method be on the fragment?

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (g-harel) [reply src/org/thoughtcrime/securesms/NicknameHandler.java]: `setupDatabaseHandler` is being called at the start of all the other methods. This means (in my understanding) that it does not need to be called from outside the `NicknameHandler` class and can be made private.

File: test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java
Comment (g-harel) [reply test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java]: Still there

File: test/unitTest/java/org/thoughtcrime/securesms/nickname/NicknameHandlerTests.java
Comment (g-harel) [inline test/unitTest/java/org/thoughtcrime/securesms/nickname/NicknameHandlerTests.java]: ...OldName*

File: test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java
Comment (veken1199) [reply test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java]: we will need to talk to @Dan-Yoo first,

**Review by g-harel — DISMISSED** [review]
What the heck was that last commit, you changed almost 2k lines !?

There are also some un-addressed comments remaining

File: test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java
Comment (veken1199) [reply test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java]: if we remove it @Dan-Yoo will spend time trying to understand and re designing the wheel. in this case, he will right away go to the solution.

File: test/unitTest/java/org/thoughtcrime/securesms/nickname/NicknameHandlerTests.java
Comment (veken1199) [reply test/unitTest/java/org/thoughtcrime/securesms/nickname/NicknameHandlerTests.java]: aghh

File: test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java
Comment (g-harel) [reply test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java]: ... Or we can communicate and he will know what to do ?

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (veken1199) [reply src/org/thoughtcrime/securesms/NicknameHandler.java]: I still believe that line of comment doesn't hurt to keep it. lets see what the other reviewers would say @shelbaz @erinb90 @Dan-Yoo @Radu-Raicea

File: src/org/thoughtcrime/securesms/database/RecipientDatabase.java
Comment (g-harel) [inline src/org/thoughtcrime/securesms/database/RecipientDatabase.java]: Is the extra space deliberate ?

File: src/org/thoughtcrime/securesms/database/RecipientDatabase.java
Comment (g-harel) [inline src/org/thoughtcrime/securesms/database/RecipientDatabase.java]: `>` already evaluates to a boolean

File: src/org/thoughtcrime/securesms/database/RecipientDatabase.java
Comment (g-harel) [reply src/org/thoughtcrime/securesms/database/RecipientDatabase.java]: Maybe something like this. Just to make it a bit clearer.

```java
return db.update(
    this.TABLE_NAME,
    values,
        ADDRESS + " = ? AND " +
        NICKNAME + " NOT Like ? ",
    new String[] {
        address.serialize(),
        values.getAsString(NICKNAME)
    });
```

File: src/org/thoughtcrime/securesms/database/RecipientDatabase.java
Comment (g-harel) [reply src/org/thoughtcrime/securesms/database/RecipientDatabase.java]: still there

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (erinb90) [inline src/org/thoughtcrime/securesms/NicknameHandler.java]: shouldn't it be formatted like:
```
private Context           context;
private RecipientDatabase recipientDatabase;
```

File: src/org/thoughtcrime/securesms/NicknameHandler.java
Comment (erinb90) [reply src/org/thoughtcrime/securesms/NicknameHandler.java]: I think javadoc comments are pretty but I'm still conflicted since the code style guidelines say no comments... I guess keep them for now unless Dr. Shang or Jinfu says otherwise.

Comment (g-harel) [reply src/org/thoughtcrime/securesms/NicknameHandler.java]: @erinb90 it isn't about having that javadoc or not. My comment was ONLY about the first sentence.

Comment (erinb90) [reply src/org/thoughtcrime/securesms/NicknameHandler.java]: Oh yeah I agree the first sentence isn't needed.

File: src/org/thoughtcrime/securesms/database/RecipientDatabase.java
Comment (erinb90) [inline src/org/thoughtcrime/securesms/database/RecipientDatabase.java]: needs the spaces before the =

**Review by g-harel — APPROVED** [review]

**Review by erinb90 — APPROVED** [review]
👍

**Pull request merged**
